app/src/main/java/io/appium/uiautomator2/model/XPathFinder.java

```diff
@@ -235,8 +236,17 @@ public static AccessibilityNodeInfo getRootAccessibilityNode() throws UiAutomato
 
     long end = SystemClock.uptimeMillis() + timeoutMillis;
     while (true) {
```

Comment:
could you also change the loop condition from `true` to `end - SystemClock.uptimeMillis() >= 0 `
This will make the readability better. Exception throwing could be moved after the loop then.